cradle-core/src/main/java/com/exactpro/cradle/messages/StoredMessageBatch.java

```diff
@@ -139,6 +137,13 @@ public long getBatchSize()
 	 * @throws CradleStorageException if message cannot be added to the batch due to verification failure or if batch limit is reached
 	 */
 	public StoredMessage addMessage(MessageToStore message) throws CradleStorageException
+	{
+		int messageSize = calculateSizeAndCheckConstraints(message);
+		
+		return addMessageInternal(message, messageSize);
+	}
+
+	protected int calculateSizeAndCheckConstraints(MessageToStore message) throws CradleStorageException
 	{
 		//first message in the batch. Need to fill it by services
 		if (messages.isEmpty()) {
```

Comment:
If first message is too large to fit in batch, it will be rejected but `batchSize` becomes assigned like message was added.